src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -51,6 +52,37 @@ public static LoginData performUserLogin() {
         return loginData;
     }
 
+    public static LoginData performAutoLogin(StartupParams params) {
+        Credentials credentials = loadCredentials();
+        try {
+            credentials.decrypt(params.getMasterPassword());
+        } catch (Exception e) {
+            System.err.println("Couldn't login, check your master password txt file");
+            e.printStackTrace();
+        }
+
+        Credentials.User user = credentials.getUsers()
+                .stream()
+                .filter(usr -> usr.u.equals(params.getUsername()))
+                .findFirst()
+                .orElse(null);
+        if (user == null) {
+            System.err.println("Invalid program arguments, make sure you entered the correct username and filepath");
+            System.exit(0);
+        }
+
+        LoginData loginData = new LoginData();
+        loginData.setCredentials(user.u, user.p);
+
+        System.out.println("Auto logging in (1/2)");
+        usernameLogin(loginData);
+        System.out.println("Loading spacemap (2/2)");
+        findPreloader(loginData);
+
+        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) System.exit(0);
```

Comment:
```suggestion
        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) {
          System.err.println("Could not find preloader url or parameters");
          System.exit(0);
        }
```
Before any random system.exit, log it. If any user reports "it's randomly closing" we need to see why in the log